scheduler/scheduler.go

```diff
@@ -231,7 +231,18 @@ func (s *EtcdScheduler) ResourceOffers(
 		memWanted   = s.memPerTask + executorWantsMem
 		portsWanted = uint64(portsPerTask + executorWantsPorts)
 	)
+
 	for _, offer := range offers {
+		// Reject offer if there's nothing to schedule.
+		s.mut.RLock()
+		if len(s.running) == s.desiredInstanceCount {
```

Comment:
would be great to ALSO check this before we ever enter the `for` loop (lock, check instance count, unlock, then reject all offers if needed), since most of the time etcd-mesos won't be accepting any offers (on a stable cluster).

Also, it's probably more efficient to decline all the offers at once (in the proposed check before the `for` loop) vs sending 1 decline call per-offer. suspect that mesos (master) supports bulk decline since v0.26 via https://issues.apache.org/jira/browse/MESOS-3522; will confirm with core team

bulk decline would be executed via `driver.LaunchTasks(allOfferIDs, []*mesos.TaskInfo{}, filters)`